system/lib/libc/crt1_proxy_main.c

```diff
@@ -49,5 +51,10 @@ EMSCRIPTEN_KEEPALIVE int _emscripten_proxy_main(int argc, char** argv) {
   pthread_t thread;
   int rc = pthread_create(&thread, &attr, _main_thread, NULL);
   pthread_attr_destroy(&attr);
+  if (rc == 0) {
+    // Mark the thread as strongly referenced, so that Node.js doesn't exit
+    // while the pthread is running.
+    _emscripten_thread_set_strongref(thread);
```

Comment:
I guess one alternative to adding this new API would be to add a custom pthread attr to signal this?    However, since this is an internal API we can always experiment with changing that later.

Reply:
I experimented with a custom pthread attr (similar to `emscripten_pthread_attr_settransferredcanvases()`), but that was more complicated than doing this. `_emscripten_thread_set_strongref()` is also used for crashed threads, so it can't be removed in favor of a custom pthread attr.